CIME/build.py

```diff
@@ -755,7 +755,7 @@ def _build_libraries(
     if mpilib == "mpi-serial":
         libs.insert(0, mpilib)
 
-    if uses_kokkos(case) and comp_interface == "mct":
+    if uses_kokkos(case):
```

Comment:
If you just remove it, does that break something in CESM?  Maybe we need "comp_interface="mct OR moab" "

Reply:
Thanks @jgfouca . Rob is correct that removing this will break the build of Kokkos through the CESM workflow. It is better to use Rob's suggested check or `comp_interface != "nuopc"` (CESM3 only uses NUOPC coupler and I assume E3SM won't use NUOPC in the future).

Reply:
@sjsprecious , that's a good idea. I have changed the code to check for nuopc.

Reply:
That's a good thought. I think a simple workaround is fine for now.